### PR TITLE
Move storage/integrity_checker -> kvbc/tools/db_integtiry_check.

### PR DESCRIPTION
**Pull request by toly-kournik**
* **Problem Overview**  
 Fix hardcoded "concord" path prefix.
Use boost::program_options to handle command line arguments. (will be used in db_restore also)

* **Testing Done**  
  manual tests with/without s3 path prefix

### CODE REVIEW

File: kvbc/tools/db_integrity_check/integrity_checker.cpp [hunk 1]
```diff
@@ -24,97 +22,95 @@
...
+    std::exit(1);
   }
 }
+void IntegrityChecker::initS3Config(const fs::path& s3_file) {
+  const std::string checkpoints_suffix("metadata/checkpoints/");
+  auto s3_config = storage::s3::ConfigFileParser(s3_file).parse();
+  LOG_DEBUG(logger_, "s3 configuration: " << s3_config);
+  dbset_ = std::make_unique<S3StorageFactory>(std::string("not used"), s3_config)->newDatabaseSet();
+  checkpoints_prefix_ =
+      s3_config.pathPrefix.empty() ? checkpoints_suffix : s3_config.pathPrefix + "/" + checkpoints_suffix;
```
Comment (WildFireFlum) [inline kvbc/tools/db_integrity_check/integrity_checker.cpp:100]: if you are using std::fs, why not use the platform independent path concat?

File: kvbc/tools/db_integrity_check/integrity_checker.cpp [hunk 2]
```diff
@@ -24,97 +22,95 @@
...
-  std::string checkpoints_prefix("concord/metadata/checkpoints/");
-  auto [key, val] = it->seekAtLeast(Sliver::copy(checkpoints_prefix.data(), checkpoints_prefix.length()));
+  auto [key, val] = it->seekAtLeast(Sliver::copy(checkpoints_prefix_.data(), checkpoints_prefix_.length()));
   (void)val;
   if (it->isEnd()) throw std::runtime_error("no checkpoints information in S3 storage");
 
-  std::string suff = key.toString().substr(checkpoints_prefix.length());
-  BlockId block = concord::util::to<BlockId>(suff.substr(0, suff.find_last_of('/')));
+  std::string suff = key.toString().substr(checkpoints_prefix_.length());
+  BlockId block = util::to<BlockId>(suff.substr(0, suff.find_last_of('/')));
```
Comment (WildFireFlum) [inline kvbc/tools/db_integrity_check/integrity_checker.cpp:113]: maybe replace with fs::path::preferred_separator?

File: kvbc/tools/db_integrity_check/integrity_checker.cpp [hunk 3]
```diff
@@ -144,15 +140,14 @@ DescriptorOfLastStableCheckpoint IntegrityChecker::getCheckpointDescriptor(const
...
+  auto key_val = it->seekAtLeast(Sliver::copy(checkpoints_prefix_.data(), checkpoints_prefix_.length()));
   if (it->isEnd()) throw std::runtime_error("no checkpoints information in S3 storage");
 
   BlockId first_good_block_descriptor = 0;
   Sliver first_good_block_descriptor_key;
   while (!it->isEnd()) {
-    std::string suff = key_val.first.toString().substr(checkpoints_prefix.length());
-    BlockId block = concord::util::to<BlockId>(suff.substr(0, suff.find_last_of('/')));
+    std::string suff = key_val.first.toString().substr(checkpoints_prefix_.length());
+    BlockId block = util::to<BlockId>(suff.substr(0, suff.find_last_of('/')));
```
Comment (WildFireFlum) [inline kvbc/tools/db_integrity_check/integrity_checker.cpp:150]: same

**Review by WildFireFlum — APPROVED** [review]
LGTM.
Would be nice if you could use std::fs instead of the hard-coded path separators.

File: kvbc/tools/db_integrity_check/integrity_checker.cpp (hunk 3 above)
Comment (toly-kournik) [reply kvbc/tools/db_integrity_check/integrity_checker.cpp:150]: same

**Pull request merged**